os/kernel/log_dump/log_dump.c

```diff
@@ -118,12 +124,30 @@ int log_dump_init(void)
 		lldbg("memory allocation failure\n");
 		return LOG_DUMP_MEM_FAIL;
 	}
+
+	/* The buffers allocated below are required for log-dump till the 
+	 * device is shut down or restarted. So these buffers MUST NEVER
+	 * BE FREED. 
+	 */
 	uncomp_array = (signed char *)kmm_malloc(CONFIG_LOG_DUMP_CHUNK_SIZE);
 	if (uncomp_array == NULL) {
 		lldbg("memory allocation failure\n");
-		kmm_free(node);
-		return LOG_DUMP_MEM_FAIL;
+		goto exit_with_node;
+	}
+
+	out_buf = allocate_compress_buffer(0, CONFIG_LOG_DUMP_CHUNK_SIZE);
+	if (out_buf == NULL) {
+		lldbg("memory allocation failure\n");
+		goto exit_with_uncomp;
+	}
+
+	last_comp_block = allocate_compress_buffer(4, CONFIG_LOG_DUMP_CHUNK_SIZE);
+	if (last_comp_block == NULL) {
+		lldbg("memory allocation failure\n");
+		goto exit_with_outbuf;
```

Comment:
We are making allocations at init time itself to avoid semaphore contentions later?